spec.bs

```diff
@@ -719,6 +719,233 @@ A [=tokenizer=] has an associated <dfn for=tokenizer>code point</dfn>, a Unicode
+  <dt>"<dfn for=part/modifier>`one-or-more`</dfn>"</dt>
+  <dd>The [=part=] has a "one or more" [=part/modifier=] indicated by the U+002B (`+`) code point.</dd>
+</dl>
+
+<h4 id=options-header>Options</h4>
+
+An <dfn>options</dfn> [=struct=] contains different settings that control how [=/pattern string=] behaves.  These options originally come from [path-to-regexp](https://github.com/pillarjs/path-to-regexp).  Not all are utilized when constructing a {{URLPattern}}, but they are maintained here to make it easier to stay aligned with [path-to-regexp](https://github.com/pillarjs/path-to-regexp).
+
+An [=/options=] has an associated <dfn export for=options>delimiter list</dfn>, a string, initially "`/#?`".  Each code point in the [=options/delimiter list=] is treated as a segment separator.  This is used, for example, for determining how far a `:foo` named group should match by default.
+
+An [=/options=] has an associated <dfn for=options>prefix list</dfn>, a string, initially "`./`".  Each code point in the [=options/prefix list=] is treated as an automatic prefix if found immediately preceding a match group.  This matters when a match group is modified to be optional or repeating.  For example, "`/foo/:bar?/baz`" will by default treat the "`/`" before "`:bar`" as a prefix that becomes optional along with the named group.  So in this example the pattern would match "`/foo/baz`".
+
+An [=/options=] has an associated <dfn export for=options>sensitive</dfn> property, a boolean, initially true.  If true, matching should be case sensitive.
+
+An [=/options=] has an associated <dfn export for=options>strict</dfn> property, a boolean, initially true.  If false, the pattern will optionally match a trailing "`/`" at the end of a string.  If true this is not done.
+
+An [=/options=] has an associated <dfn export for=options>end</dfn> property, a boolean, initially true.  If true, requires matching to the end of the input string.
+
+An [=/options=] has an associated <dfn export for=options>start</dfn> property, a boolean, initially true.  If true, requires matching from the start of the input string.
+
+<h4 id=parsing>Parsing</h4>
+
+<div algorithm>
+An <dfn>encoding callback</dfn> is an abstract algorithm that takes a given string |input|.  The |input| will be a simple text piece of a [=/pattern string=].  An implementing algorithm should validate and encode the |input|.  It must return the encoded string or throw an exception.
+</div>
+
+A <dfn>pattern parser</dfn> is a [=struct=].
+
+A [=pattern parser=] has an associated <dfn for="pattern parser">token list</dfn>, a [=/token list=], initially an empty [=list=].
+
+A [=pattern parser=] has an associated <dfn for="pattern parser">encoding callback</dfn>, a [=/encoding callback=], initially null.
+
+A [=pattern parser=] has an associated <dfn for="pattern parser">options</dfn>, a [=/options=], initially null.
```

Comment:
Optional, but perhaps a bit clearer: if there are mandatory fields that are always set upon creation, you can say "which must be set upon creation". Then when creating you can say something like "a new pattern parser whose encoding callback is X and options is Y".

This makes it obvious which of these defaults are real defaults that are going to stick around. Right now it seems like some of these defaults are arbitrary and could be anything because you always overwrite them.

Reply:
Same goes for the part struct it seems.

Reply:
The option defaults actually come from path-to-regexp, but I can drop them since we don't use them.